project/source/Scripts/Objects/BufferSolution.gd

```diff
@@ -3,24 +3,24 @@ extends LabObject
 # Currently the same as DummySourceContainer to use for testing
 
 @export var substance: PackedScene = null
-var contents = null
+var contents:Node = null
 
-func _ready():
+func _ready() -> void:
 	super()
 	if substance == null:
 		substance = load('res://Scenes/Objects/BufferSolutionSubstance.tscn')
 	contents = substance.instantiate()
 
-func CheckContents(group):
+func CheckContents(group:String) -> bool:
 	return contents.is_in_group(group)
 
-func TakeContents():
+func TakeContents() -> Array:
```

Comment:
could probably make this return type be an Array of Nodes

Reply:
`TakeContents` seems to consistently return an array of `Substance`, and `AddContents` always takes an array of `Substance`, so I think that might be the best option for these.